router/dataplane.go

```diff
@@ -2157,23 +2195,27 @@ func (p *scionPacketProcessor) validateEgressID() (respr processResult, reserr e
 			cannotRoute,
 		)
 	}
-
 	// @ p.d.getLinkTypesMem()
 	ingress, egress := p.d.linkTypes[p.ingressID], p.d.linkTypes[pktEgressID]
 	if !p.segmentChange {
 		// Check that the interface pair is valid within a single segment.
 		// No check required if the packet is received from an internal interface.
+		// @ assert !p.segmentChange
 		switch {
 		case p.ingressID == 0:
+			// @ assume AbsValidateEgressIDConstraint(oldPkt, (p.ingressID != 0), dp)
 			// @ fold p.d.validResult(respr, false)
 			return processResult{}, nil
 		case ingress == topology.Core && egress == topology.Core:
+			// @ assume AbsValidateEgressIDConstraint(oldPkt, (p.ingressID != 0), dp)
 			// @ fold p.d.validResult(respr, false)
 			return processResult{}, nil
 		case ingress == topology.Child && egress == topology.Parent:
+			// @ assume AbsValidateEgressIDConstraint(oldPkt, (p.ingressID != 0), dp)
 			// @ fold p.d.validResult(respr, false)
 			return processResult{}, nil
 		case ingress == topology.Parent && egress == topology.Child:
+			// @ assume AbsValidateEgressIDConstraint(oldPkt, (p.ingressID != 0), dp)
```

Comment:
to separate the assumptions about IO that are temporarily here until we finish the proof of the lemmas, I would introduce a function in `defs.gobra` called `AssumeForIO` or sth like this that takes a boolean parameter and whose postcondition is that the condition holds. Later on, we can revisit the calls to this function to get a sense for what we need to prove.